Jack/Shrink.fs

```diff
@@ -97,11 +98,26 @@ module Shrink =
         if destination = x then
             LazyList.empty
         else
-            let two = LanguagePrimitives.GenericOne + LanguagePrimitives.GenericOne
+            let one : ^a = LanguagePrimitives.GenericOne
+            let two : ^a = one + one
 
             /// We need to halve our operands before subtracting them as they may be using
             /// the full range of the type (i.e. 'MinValue' and 'MaxValue' for 'Int32')
-            let diff = (x / two) - (destination / two)
+            let diff : ^a = (x / two) - (destination / two)
```

Comment:
I added these annotations so that the compiler infers them all to be the same unified type rather than generating piles of constraints